src/Server/GRPCServer.cpp

```diff
@@ -320,8 +321,20 @@ namespace
 
         Poco::Net::SocketAddress getClientAddress() const
         {
-            String peer = grpc_context.peer();
-            return Poco::Net::SocketAddress{peer.substr(peer.find(':') + 1)};
+            /// Returns a string like ipv4:127.0.0.1:55930 or ipv6:%5B::1%5D:55930
+            String uri_encoded_peer = grpc_context.peer();
+
+            if (uri_encoded_peer.starts_with("ipv4") || uri_encoded_peer.starts_with("ipv6"))
+            {
+                uri_encoded_peer = uri_encoded_peer.substr(uri_encoded_peer.find(':') + 1);
+
+                String peer;
+                Poco::URI::decode(uri_encoded_peer, peer);
+
+                return Poco::Net::SocketAddress{Poco::Net::AddressFamily::Family::IPv4, uri_encoded_peer};
```

Comment:
This line looks wrong - after you checked the prefix you know here that it's either "ipv4" or "ipv6". But the code in this line always uses `Poco::Net::AddressFamily::Family::IPv4`...

Reply:
Whoops